src/orderly/core.py

```diff
@@ -73,6 +76,70 @@ def resource(files):
     return files_expanded
 
 
+def shared_resource(files: Union[str, List[str], Dict[str, str]]):
```

Comment:
nice to support all these forms of files. Do you want to add the return type too?

Reply:
Done.